libr/cons/dietline.c

```diff
@@ -1196,12 +1250,271 @@ R_API const char *r_line_readline_cb_win(RLineReadCallback cb, void *user) {
+	for (;;) {
+		int rep = 0;
+		__print_prompt ();
+		if (!flag) {
+			break;
+		}
+		ch = r_cons_readchar ();
+		while (IS_DIGIT (ch)) {
+			int tmp = ch - '0';
+			rep = (rep * 10) + tmp;
+			ch = r_cons_readchar ();
+		}
+		rep = rep > 0? rep: 1;
+		switch (ch) {
+		case 'D':  
+			delete_till_end ();
+			break;
+		case 'r': {
+			char c =  r_cons_readchar ();
+			I.buffer.data[I.buffer.index] = c;
+		        break;
+		}
+		case 'x':
+			while (rep--) {
+				delete_next_char (); 
+				__print_prompt ();
+			} break;
+		case 'c': 
+			flag = false;
+		case 'd': {
+			int rep1 = 0;
+			char c = r_cons_readchar ();
+			while (IS_DIGIT (c) && ch != '0') {
+				int tmp = c - '0';
+				rep1 = (rep1 * 10) + tmp;
+				c = r_cons_readchar ();
+			}
+			rep1 = rep1 > 0? rep1: 1;
+			while (rep1--) {
+				switch (c) {
+				case 'W':
+					kill_Word ();
+					break;
+				case 'w':
+					kill_word ();
+					break;
+				case 'B': 
+					backward_kill_Word ();
+					break;
+				case 'b':
+					backward_kill_word ();
+					break;
+				case 'h':
+					delete_prev_char ();
+					break;
+				case 'l':
+					delete_next_char ();
+					break;
+				case '$':
+					delete_till_end ();
+					break;
+				case '0':
+					strncpy (I.buffer.data, I.buffer.data + I.buffer.index, I.buffer.length);
+					I.buffer.length -= I.buffer.index;
+					I.buffer.index = 0;
+					break;
+				} __print_prompt ();
+			}
+			} break;
+		case 'I':
+			flag = false;
+			rep = 0;
+		case '0': 
+			if (gcomp) {
+				strcpy (I.buffer.data, gcomp_line);
+				I.buffer.length = strlen (I.buffer.data);
+				I.buffer.index = 0;
+				gcomp = false;
+			}
+			I.buffer.index = 0;
+			break;
+		case 'A':
+			flag = false;
+		case '$': 
+			if (gcomp) {
+				strcpy (I.buffer.data, gcomp_line);
+				I.buffer.index = strlen (I.buffer.data);
+				I.buffer.length = I.buffer.index;
+				gcomp = false;
+			} else {
+				I.buffer.index = I.buffer.length;
+			} break;
+		case 'p': 
+			while (rep--) {
+				paste ();
+				__print_prompt();
+			} break;
+		case 'i': {
+			flag = false;
+			break;
+		}
+		case 'h': 
+			while (rep--) {
+#if USE_UTF8
+			{
+				char *s = I.buffer.data + I.buffer.index - 1;
+				utflen = 1;
+				while (s > I.buffer.data && (*s & 0xc0) == 0x80) {
+					utflen++;
+					s--;
+				}
+			}
+			I.buffer.index = I.buffer.index
+					 ? I.buffer.index - utflen
+					 : 0;
+#else
+			I.buffer.index = I.buffer.index
+					 ? I.buffer.index - 1
+					 : 0;
+#endif
+			__print_prompt();
+		} break;
+		case 'l': 
+			while (rep--) {
+#if USE_UTF8
+		{
+			char *s = I.buffer.data + I.buffer.index + 1;
+			utflen = 1;
+			while ((*s & 0xc0) == 0x80) {
+				utflen++;
+				s++;
+			}
+		}
+		I.buffer.index = I.buffer.index < I.buffer.length
+			 ? I.buffer.index + utflen
+			 : I.buffer.length;
+		
+#else
+		I.buffer.index = I.buffer.index < I.buffer.length
+			? I.buffer.index + 1
+			: I.buffer.length;
+#endif
+			__print_prompt();
+		} break;
+		case 'B': 
+			while (rep--) {
```

Comment:
looks like a bit spaguetti and bad indent

Reply:
fixed indent.